javaslang/src/main/java/javaslang/collection/Map.java

```diff
@@ -186,9 +186,11 @@ default V apply(K key) {
     @Override
     Map<K, V> clear();
 
+    @SuppressWarnings("unchecked")
     @Override
-    default boolean contains(Tuple2<K, V> element) {
-        return get(element._1).map(v -> Objects.equals(v, element._2)).getOrElse(false);
+    default boolean contains(Object element) {
+        Tuple2<K, V> tuple = (Tuple2<K, V>) element;
```

Comment:
but <code>contains(Tuple2<K, V> element)</code> not compiles because it override <code>contains(Object element)</code>